### PR TITLE
Adding features SignupLogin-DemoUser-ProtectedRoute-#12-9-14

### PR DESCRIPTION
**Pull request by hapalacios**
### What this PR does (required): FE: Signup / Login Pages #12
- PR functionality: this PR allows user to navigate in a login and signup pages.
- The Login and Sign up pages were matched to the mockups provided.

### Screenshots / Videos (required):
- Login page:
Screen Shot 2021-08-05 at 4 11 32 PM

- Signup page:
Screen Shot 2021-08-05 at 4 11 40 PM

### Any information needed to test this feature (required):
- To test these features/pages:
(a) Npm start to start the page in localhost, and then click in the "Create/Login" button at the bottom of the page to navigage between pages.
(b) Write directly in the browser address bar: http://localhost:3000/login and/or http://localhost:3000/signup

###########################################################################

### What this PR does (required): Demo User #9
- PR functionality: this feautre allows users to navigate using a demo user.
- This will allow recruiters and managers to easily test the app's features without having to sign up \
- To do this and "Add" button was added to allow for signing in as a demo user.
- The button is both, in "Login" and "Sign up" pages.

### Screenshots / Videos (required):
- Step 1: demo button is clicked
Screen Shot 2021-08-05 at 5 24 21 PM

- Result: user redirected to test landing page http://localhost:3000/dashboard 
Screen Shot 2021-08-05 at 5 24 30 PM

### Any information needed to test this feature (required):
- From Login or Sign up page, just click "Demo Button".

### Any issues with the current functionality (optional):
- No issues.
- Ticket related with # 9 and # 14, demo user and protected route. For that reason were  submitted together.

###########################################################################

### What this PR does (required): FE/IN: Protected Route #14
- PR functionality: protected route to deny access to users not login in with "demo user" button or existing users.

### Screenshots / Videos (required):
- Step 1: Write http://localhost:3000/dashboard without using "Demo user" button before.

- Result: user redirected to "unauthorized" page.

### Any information needed to test this feature (required):
- Step 1: Write http://localhost:3000/dashboard without using "Demo user" button before.
Screen Shot 2021-08-05 at 6 15 07 PM
- Result: user redirected to "unauthorized" page.

### Any issues with the current functionality (optional):
- No issues.
-  Disabled redirection in Authcontext.tsx: history.push('/login'). This should be enabled again once server and authContext is working again.
- Ticket related with # 9 and # 14, demo user and protected route. For that reason were  submitted together.

### Video: 
https://user-images.githubusercontent.com/60803185/128429502-bc73af1d-176b-419d-b4db-94aed12478ad.mp4

### CODE REVIEW

File: client/src/components/ProtectedRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      {...rest}
+      render={(props) => {
+        if (loggedInUser === undefined || !demoUser) {
+          return (
+            <Redirect
+              to={{
+                pathname: '/unauthorized',
+                state: {
+                  from: props.location,
+                },
```
Comment (moffatethan) [inline client/src/components/ProtectedRoute.tsx:26]: We don't need an unauthorized page just redirect to login

**Review by moffatethan — APPROVED** [review]
I'll approve, please be careful with line sequencing as it causes rather large PRs

**Comment by moffatethan** [comment]
**Please resolve conflicts before merging**

File: client/src/components/ProtectedRoute.tsx (hunk 1 above)
Comment (hapalacios) [reply client/src/components/ProtectedRoute.tsx]: Done. Thank you!

**Comment by hapalacios** [comment]
Dashboard.jxs and styling were accepted from current feature branch. This should replaced later with the Dashboard feature.

Package.json and lock.json were accepted from origin (main). 

4 file conflicts resolved.

**Pull request merged**